core/tee/tee_svc_storage.c

```diff
@@ -453,9 +468,9 @@ TEE_Result syscall_storage_obj_create(unsigned long storage_id, void *object_id,
 		goto err;
 	}
 
-	o->info.handleFlags =
-	    TEE_HANDLE_FLAG_PERSISTENT | TEE_HANDLE_FLAG_INITIALIZED;
-	o->flags = flags;
+	o->info.handleFlags = TEE_HANDLE_FLAG_PERSISTENT |
+			      TEE_HANDLE_FLAG_INITIALIZED |
+			      (flags & ~TEE_DATA_FLAG_OVERWRITE);
```

Comment:
Why clear `TEE_DATA_FLAG_OVERWRITE`?

Reply:
Once the object is created this flag is of no interest.
But, above all it's because some of the GP 1.1 tests fails if this isn't cleared. :-)

Reply:
> Once the object is created this flag is of no interest.

True, not an intrinsic property of the object indeed.

> But, above all it's because some of the GP 1.1 tests fails if this isn't cleared. :-)

Very good reason :) How about adding a short comment with the GP test reference, since I could not find anything explicit in the spec itself? (such as: `/* GP 1.1 test xx-yy-zz */`).

Reply:
Hmm, I was wrong. I can't find any failing case when not clearing this flag.
I'll add a comment explaining that it's not an intrinsic property of the object as the reason why it's cleared.

Reply:
The spec for `handleFlags` in `TEE_ObjectInfo1()` mentions:

> `TEE_DATA_FLAG_XXX`: Only for persistent objects, the flags used to open or create the object.

Nothing about `TEE_DATA_FLAG_OVERWRITE` being special, so clearing it is a bit dubious I think.

Reply:
OK, I'll drop that part then.